express/tests/validators/emailValidator.test.ts

```diff
@@ -0,0 +1,37 @@
+import {expect} from "chai";
+import {validateEmail} from "../../src/lib/emailValidator";
+
+const EMAIL_WITH_ALLOWED_DOMAIN = "allowed@email.gov.uk";
+const EMAIL_WITH_DISALLOWED_DOMAIN = "allowed@hackers.co.uk";
+
+const EMAIL_WITH_NO_AT_SYMBOL = "copyAndPaste_at_some.domain.gov.uk";
+const EMAIL_COMPLYING_WITH_RFC822 = "an_Email+modification@some.domain.gov.uk";
+
+describe("Checking that the email domain is on the allowed list in 'allowed-email-domains.txt'", () => {
+    it("should return true for an email with a domain in the list of allowed domains", async () => {
+        expect(await validateEmail(EMAIL_WITH_ALLOWED_DOMAIN)).to.eql({isValid: true}, `${EMAIL_WITH_ALLOWED_DOMAIN} should be allowed`);
```

Comment:
In a future iteration we need to get rid of those awaits in the validators. We should pre-load the domains and other resources at application startup.